dtests/src/test/scala/io/snappydata/hydra/TestUtil.scala

```diff
@@ -58,15 +60,18 @@ object TestUtil {
    validation has failed.
    */
   def assertQuery(snc: SnappyContext, sqlString: String, numRows: Int, queryNum: String, pw:
-  PrintWriter, sqlContext: SQLContext): Boolean = {
+  PrintWriter, sqlContext: SQLContext, skipNumRowsValidation: Boolean): Boolean = {
+
     var hasValidationFailed = false
-    val df = snc.sql(sqlString)
-    pw.println(s"No. rows in resultset for query ${queryNum} is : ${df.count} for ${tableType} table")
-    if (df.count() != numRows) {
-      pw.println(s"Result mismatch for query ${queryNum} : found ${df.count} rows but expected ${numRows} rows.")
-      hasValidationFailed = true
+    if (!skipNumRowsValidation) {
```

Comment:
The 'assertJoin' method is very much similar to 'assertQuery' method ,it will be nice ,if one of the method is reused.

Reply:
I think Sonal has already done this in her code refactoring changes in end-to-end framework validation branch, so I did not change this original code.